### PR TITLE
Adding SearchParameter for QuestionnaireResponse

### PR DESCRIPTION
**Pull request by yuliumitre**
This PR adds search parameter "context" for QR. 
To test:
1. Create QR with context: e.g. save in-progress QR to EHR
2. Search the QR via postman with a query, 
  e.g.:  http://localhost:8080/test-ehr/r4/QuestionnaireResponse?context=DeviceRequest/devreq014
3. The returned bundle should contain the matching QR

### CODE REVIEW

**Comment by KeeyanGhoreshi** [comment]
I can't seem to get this to work, request to the endpoint returns `Unknown search parameter \"context\" for resource type \"QuestionnaireResponse\"`

**Comment by yuliumitre** [comment]
> I can't seem to get this to work, request to the endpoint returns `Unknown search parameter \"context\" for resource type \"QuestionnaireResponse\"`

What do you get if you run the following query? 
http://localhost:8080/test-ehr/r4/SearchParameter

**Comment by KeeyanGhoreshi** [comment]
I receive the following bundle:

```
{
    "resourceType": "Bundle",
    "id": "5cde0eab-6447-4e6c-a3d6-f05ca11dcc49",
    "meta": {
        "lastUpdated": "2022-08-12T14:25:32.270-04:00"
    },
    "type": "searchset",
    "total": 1,
    "link": [
        {
            "relation": "self",
            "url": "http://localhost:8080/test-ehr/r4/SearchParameter"
        }
    ],
    "entry": [
        {
            "fullUrl": "http://localhost:8080/test-ehr/r4/SearchParameter/852",
            "resource": {
                "resourceType": "SearchParameter",
                "id": "852",
                "meta": {
                    "versionId": "1",
                    "lastUpdated": "2022-08-12T13:33:49.637-04:00",
                    "source": "#u9y6uQ0t4482DHBc"
                },
                "status": "active",
                "base": [
                    "QuestionnaireResponse"
                ],
                "type": "reference",
                "expression": "QuestionnaireResponse.extension('http://hl7.org/fhir/us/davinci-dtr/StructureDefinition/context')"
            },
            "search": {
                "mode": "match"
            }
        }
    ]
}
```

File: src/main/resources/fhirServer.properties [hunk 1]
```diff
@@ -2,8 +2,8 @@ client_id = app-token
 client_secret= #replaceMe#
 realm=ClientFhirServer
 use_oauth = false
-oauth_token = http://localhost:8180/auth/realms/ClientFhirServer/protocol/openid-connect/token
-oauth_authorize =  http://localhost:8180/auth/realms/ClientFhirServer/protocol/openid-connect/auth
+oauth_token = http://localhost:8180/realms/ClientFhirServer/protocol/openid-connect/token
```
Comment (KeeyanGhoreshi) [inline src/main/resources/fhirServer.properties:5]: this reintroduced the keycloak auth changes

**Review by KeeyanGhoreshi — APPROVED** [review]
LGTM

**Pull request merged**